thriftbp/client_pool.go

```diff
@@ -62,9 +62,23 @@ type ClientPoolConfig struct {
 	// "${host}:${port}"
 	Addr string `yaml:"addr"`
 
-	// InitialConnections is the inital number of thrift connections created by
-	// the client pool.
-	InitialConnections int `yaml:"initialConnections"`
+	// InitialConnections is the desired inital number of thrift connections
+	// created by the client pool.
+	//
+	// If an error occurred when we try to establish the initial N connections for
+	// the pool, we log the error with InitialConnectionsFallbackLogger,
+	// then return the pool with the <N connections we already established.
+	//
+	// If that's unacceptable, then RequiredInitialConnections can be used to set
+	// the hard requirement of minimal initial connections to be established.
+	// Note that enabling this can cause cascading failures during an outage,
+	// so it shall only be used for extreme circumstances.
```

Comment:
... or special cases like integration or load tests?

Reply:
I don't think it should even be used in these 2 cases. Say you set `RequiredInitialConnections` to 1 in a (integration|load) test and the first client failed, that does not necessarily mean the remote server is just unavailable. it could well be just a dns blip, or any network blip. The test shall only fail in cases of, say, the failure rate is higher than some threshold. The first client connection attempt failed does not really map to the failure rate threshold in any way.